### PR TITLE
Pci 1809 fix python issues

### PR DESCRIPTION
**Pull request by aliculPix4D**
- adds regression test for multiple Python dependencies and files
- fixes the issue by always passing all the input files to the second iteration, but update files only when needed

### CODE REVIEW

File: pix4d-dependabot/spec/helpers/helper_dependabot_spec.rb [hunk 1]
```diff
@@ -293,3 +293,74 @@ def fixture(package_manager, *name)
     end
   end
 end
+
+RSpec.describe "describe dependencies_updater function", :pix4d do
```
Comment (aliculPix4D) [inline pix4d-dependabot/spec/helpers/helper_dependabot_spec.rb:297]: This is regression test

**Comment by Pix4D-Janus** [comment]
Your pipeline is set.
Check it here:https://builder.ci.pix4d.com/teams/developers/pipelines/dependabot-PCI-1809-fix-python-issues

**Review by odormond — APPROVED** [review]

**Review by Esysc — APPROVED** [review]

**Comment by marco-m-pix4d** [comment]
Looking at the commit comments it is not clear to understand what this regression test is about.
I remember we discussed adding a regression test, but for what?
Can we come up with a clear and concise description, or a link to a previous PR or ticket, and add it to the commit comments?

**Review by marco-m-pix4d — APPROVED** [review]
I am pre-approving, but please have a look at my comment. Merci.

**Review by manuel-pix4d — APPROVED** [review]
It fixes the issue. Good job ! As pointed by Marco, we need some comments to understand what does the test. And all the main functionalities, that handles different pypi with different credentials are not tested.  But I don't think it would be useful to test it in unit test. This is more for integration tests, where we cover the work case of Pix4D usage. Definitely not in this PR.

**Comment by aliculPix4D** [comment]
> Can we come up with a clear and concise description, or a link to a previous PR or ticket, and add it to the commit comments?

Done as suggested.

**Pull request merged**